cmd/traffic/cmd/agent/fwdstate.go

```diff
@@ -94,48 +111,217 @@ func (pm *ProviderMux) CreateClientStream(ctx context.Context, tag tunnel.Tag, s
+
+		if isSubset1of2 || isSubset2of1 {
+			if !hasPaths1 || !hasPaths2 || pathFiltersOverlap(spec1.PathFilters, spec2.PathFilters) {
+				subsetDesc := "header filters form a subset"
+				if hasPaths1 && hasPaths2 {
+					subsetDesc += " and paths overlap"
+				}
+				return subsetDesc
 			}
 		}
-		// Disjoint path filters = no conflict
-		return false
 	}
 
-	// If one has header filters and the other has path filters, they can coexist
-	// Each will only match traffic meeting their specific criteria
-	if (len(spec1.HeaderFilters) > 0 && len(spec2.PathFilters) > 0) ||
-		(len(spec1.PathFilters) > 0 && len(spec2.HeaderFilters) > 0) {
+	if hasPaths1 && hasPaths2 && !hasHeaders1 && !hasHeaders2 {
+		return "path filters overlap"
+	}
+
+	return "filters would route the same traffic to different destinations"
+}
+
+// isHeaderSubset checks if all headers in subset exist in superset with matching values.
+// Uses wildcard matching for header values.
+func isHeaderSubset(subset, superset map[string]string) bool {
+	for key, value := range subset {
+		if superValue, exists := superset[key]; !exists || !headerValueMatches(value, superValue) {
+			return false
+		}
+	}
+	return true
+}
+
+// interceptSpecsConflict determines if two intercept specs would conflict with each other.
+// Two specs conflict if they would route the same traffic to different destinations.
+//
+// Precedence Model:
+// Headers take precedence over paths. This means intercepts with headers operate at a higher
+// priority tier than intercepts with only paths.
+//
+// Conflict Rules:
+// 1. Global intercepts (no headers, no paths) conflict with everything
+// 2. Headers vs Paths: One spec with headers, another with only paths → NO CONFLICT
+//    (different priority tiers - headers are checked first, then paths)
+// 3. Both have headers: Conflict if headers form a subset AND paths overlap
+//    - Within each intercept, filters use AND logic (must match ALL headers AND ALL paths)
+//    - Example: {x-user:adam} vs {x-user:adam, x-session:xyz} → CONFLICT (first is subset)
+//    - Example: {x-user:adam}+/api/* vs {x-user:adam}+/admin/* → NO CONFLICT (different paths)
+// 4. Both have only paths (no headers): Conflict if paths overlap
+func interceptSpecsConflict(spec1, spec2 *manager.InterceptSpec) bool {
+	hasHeaders1 := len(spec1.HeaderFilters) > 0
+	hasHeaders2 := len(spec2.HeaderFilters) > 0
+	hasPaths1 := len(spec1.PathFilters) > 0
+	hasPaths2 := len(spec2.PathFilters) > 0
+
+	// Global intercept: no headers and no paths means it intercepts everything
+	isGlobal1 := !hasHeaders1 && !hasPaths1
+	isGlobal2 := !hasHeaders2 && !hasPaths2
+
+	// Rule 1: Global intercepts conflict with anything
+	if isGlobal1 || isGlobal2 {
+		return true
+	}
+
+	// Rule 2: Headers take precedence - different priority tiers don't conflict
+	// If one spec has headers and the other has only paths, they operate at different tiers
+	if hasHeaders1 && !hasHeaders2 && !isGlobal2 {
```

Comment:
`isGlobal1` and `isGlobal2` are both guaranteed to be `false` for the remainder of this function.